test/test_torch.py

```diff
@@ -2902,60 +2892,91 @@ def test_tensor_factory_cuda_type(self):
         self.assertTrue(x.is_cuda)
         torch.set_default_tensor_type(saved_type)
 
-    # This is a temporary test for a boolean tensors on CPU. Once the CUDA part
-    # will be done, these test cases will be moved down to test_tensor_factories_empty test
-    def test_tensor_factories_bool(self):
-        expectedShape = (1, 2)
-        test = torch.empty(expectedShape, dtype=torch.bool)
-        self.assertEqual(expectedShape, test.shape)
-
-        test2 = torch.empty_like(test, dtype=torch.bool)
-        self.assertEqual(test.shape, test2.shape)
-
-        test = torch.full(expectedShape, True, dtype=torch.bool)
-        self.assertEqual(test, torch.tensor([[True, True]], dtype=torch.bool))
-
-        test2 = torch.full_like(test, True, dtype=torch.bool)
-        self.assertEqual(test, test2)
-
-        test = torch.zeros(expectedShape, dtype=torch.bool)
-        self.assertEqual(test, torch.tensor([[False, False]], dtype=torch.bool))
-
-        test2 = torch.zeros_like(test, dtype=torch.bool)
-        self.assertEqual(test, test2)
-
-        test = torch.ones(expectedShape, dtype=torch.bool)
-        self.assertEqual(test, torch.tensor([[True, True]], dtype=torch.bool))
-
-        test2 = torch.ones_like(test, dtype=torch.bool)
-        self.assertEqual(test, test2)
+    def test_unfolf_all_devices_and_dtypes(self):
+        for device in torch.testing.get_all_devices():
+            for dt in torch.testing.get_all_dtypes(True):
+                if dt == torch.half and device == 'cpu':
+                    # fix once random is implemented for Half on CPU
+                    self.assertRaises(RuntimeError, lambda: torch.randint(5, (0, 1, 3, 0), dtype=dt, device=device))
+                else:
+                    x = torch.randint(5, (0, 1, 3, 0), dtype=dt, device=device)
+                    self.assertEqual((0, 1, 1, 0, 3), x.unfold(2, 3, 2).shape)
 
-        test = torch.randint(10, expectedShape, dtype=torch.bool)
-        self.assertEqual(expectedShape, test.shape)
-        self.assertEqual(torch.bool, test.dtype)
+    def test_copy_all_dtypes_and_devices(self):
+        from copy import copy
+        for device in torch.testing.get_all_devices():
+            for dt in torch.testing.get_all_dtypes(True):
+                x = torch.tensor([1, 2, 3, 4], dtype=dt, device=device)
+                y = copy(x)
+
+                # copy is a shallow copy, only copies the tensor view,
+                # not the data
```

Comment:
Umm... you've lost the most important part of the test here, which is the fill on `x` should propagate to `y`. Re-check the old implementation.

Reply:
(`assertEqual` doesn't do reference equality check; it checks if it's element-by-element the same)